apps/roam/src/utils/internalError.ts

```diff
@@ -0,0 +1,69 @@
+import posthog from "posthog-js";
+import type { Properties } from "posthog-js";
+import renderToast from "roamjs-components/components/Toast";
+import { getVersionWithDate } from "~/utils/getVersion";
+import getCurrentUserDisplayName from "roamjs-components/queries/getCurrentUserDisplayName";
+import sendErrorEmail from "~/utils/sendErrorEmail";
+
+const internalError = ({
+  error,
+  userMessage,
+  type,
+  context,
+  sendEmail, // true by default
+}: {
+  error: unknown;
+  type?: string;
+  userMessage?: string;
+  context?: Properties;
+  sendEmail?: boolean;
+}): void => {
+  if (process.env.NODE_ENV === "development") {
+    console.error(error, context);
+  } else {
+    const { version, buildDate } = getVersionWithDate();
+    const username = getCurrentUserDisplayName();
+    if (username) posthog.identify(username);
+    type = type || "internal-error";
+    context = {
+      app: "Roam",
+      type,
+      graphName: window.roamAlphaAPI?.graph?.name || "unknown",
+      version: version || "-",
+      buildDate: buildDate || "-",
+      ...(context || {}),
+    };
+
+    if (typeof error === "string") {
+      error = new Error(error);
+    } else if (!(error instanceof Error)) {
+      try {
+        const serialized = JSON.stringify(error);
+        error = new Error(serialized);
+      } catch {
+        error = new Error(typeof error);
+      }
+    }
+    posthog.captureException(error, context);
+    if (sendEmail !== false) {
+      sendErrorEmail({
+        // by now error is an Error but TS did not figure it out
+        error: error as Error,
+        type,
```

Comment:
Not really a big fan of having the default email subject/headers changed.  This changes the grouping and search functionality.  Could we please keep the existing functionality?  If you'd like to add a specific "slug" for posthog, feel free add that/convert the existing titles.